### ISSUE TITLE
Change the app's icon

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#868]
It seems the app's icon needs some update. It doesn't look so nice on some backgrounds.

![commons logo](https://user-images.githubusercontent.com/12448084/30127922-8098d98c-935e-11e7-9c82-512b8858cc82.png)

Though the logo seems to be clear in the above screen shot it's not when the brightness is low. It's very hard to identify it.

How about giving it a circular (Android O's coming) white background to it.

**Comment by tim-weaver** [comment]
i gave it a white border

**Comment by psh** [comment]
If we're going to play around with the icon, I'd suggest digging into Google's style guide - https://material.io/guidelines/style/icons.html

The other small change would be to create 'adaptive icons' (see https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive.html) so that Oreo users see icons more suited to the platform.

**Comment by thedeadline** [comment]
Hi, 
I'd like to create an adaptive icon for the app but I can't find the raw file for the icon... is the original vector file (illustrator, sketch, etc.) available in the repo? 

Thanks, I'm new to open source contribution (doing it for Hacktoberfest) :).

**Comment by nicolas-raoul** [comment]
The SVG is on Wikimedia Commons 🙂

**Comment by psh** [comment]
@nicolas-raoul - Do you think it would be a good idea to drop a copy into the 'design' folder of this project, just for reference?

**Comment by thedeadline** [comment]
I agree with @psh I thought it would be in 'design' and looked there first.

**Comment by nicolas-raoul** [comment]
Here is the SVG: https://commons.wikimedia.org/wiki/File:Commons-logo.svg
Yes, feel free to add it inside the project, good idea!

**Comment by neslihanturan** [comment]
Actually a related PR already merged, but our logo still looks same, I couldn't understand the reason. But if we are okay with current look of the icon, I will close this discussion. Personally, I am okay.

**Comment by nicolas-raoul** [comment]
The merge was only so that it behaves differently on Android O, apparently.

I agree that the icon is sometimes not very visible. If anyone has a good design that make the Commons icon more visible, feel free to propose it.

**Comment by neslihanturan** [comment]
Actually @tim-weaver made white borders for our icon at #875 but since the PR didn't merged for long time, and I think it doesn't look very beautiful with white borders, closed the PR. We can always re-open and merge it if you like white borders. But I think we are still welcome design ideas for Commons icon.

**Comment by sivaraam** [comment]
> We can always re-open and merge it if you like white borders. 

I think it doesn't improve things much.

> But I think we are still welcome design ideas for Commons icon.

How about using one version of the adaptive icons (I like the square one with the rounded corners) for other android versions? I guess it would solve the issue.

**Comment by whym** [comment]
It might be useful to ask Wikimedia Foundation Legal team about what kind of modifications are allowed to the original logo which is a trademark of theirs. (I believe this app's logo is under a trademark agreement with them, right? Unless perhaps we choose an entirely new one.)

I found https://wikimediafoundation.org/wiki/Visual_identity_guidelines but it only talks bout the Wikipedia logo. (They are fairly restrictive about it, by the way)

**Comment by utsha1510** [comment]
if in case we can choose an entirely new one, I can design one.

**Comment by nicolas-raoul** [comment]
@utsha1510 It should be based on the Commons icon (SVG above).

Anyone feel free to propose an icon (or prototype of icon), but be aware that the WMF might refuse it, so please don't be upset if that happens ^_^ Cheers!

**Comment by sdabhi23** [comment]
![logo_flat](https://user-images.githubusercontent.com/25759344/44375108-92c0c700-a50f-11e8-86d8-fcaf808bd2b0.png)
![logo_elevated](https://user-images.githubusercontent.com/25759344/44375126-a79d5a80-a50f-11e8-91f0-944026c25707.png)

These are the 2 designs I could come up with, by doing minimum modifications in the original logo.

**Comment by nicolas-raoul** [comment]
I like the first one.
The concept of adding a white rounded rectangle background is also used by the Wikipedia and Wikivoyage apps, which is a plus.
I would say that I support switching to that icon.

**Comment by misaochan** [comment]
The first icon looks good to me too. :) 

@nicolas-raoul I guess we will have to ask WMF Legal first before we can use it?

**Comment by nicolas-raoul** [comment]
@misaochan Good idea indeed!

**Comment by sdabhi23** [comment]
> The merge was only so that it behaves differently on Android O, apparently.

The latest update broke it, and now it doesn't even work on Android O.

**Comment by debpedrano** [comment]
Hello! how was everyone? can I submit my design for this project?

**Comment by sdabhi23** [comment]
@debpedrano you will have to post the design on this issue. If it is approved then they will let you know about the assets they require.

**Comment by debpedrano** [comment]
@sdabhi23 
Design # 1

![image](https://user-images.githubusercontent.com/40510200/47959453-a9ab8b80-e01f-11e8-804d-5d55bde8c064.png)

**Comment by sdabhi23** [comment]
@misaochan any updates on the designs I suggested?

**Comment by misaochan** [comment]
@debpedrano We really appreciate the effort, but it is extremely unlikely that we will change the base logo, as it is based on the Wikimedia Commons logo and is a big part of our identity. However, we are open to changes to the app's **icon**, as long as the base logo remains the same. :) For instance, you can modify shading or the background color, etc.

**Comment by misaochan** [comment]
@sdabhi23 I apologize that I have not had the time to talk to WMF Legal about this due to grant deadlines. If they approve, I am good with switching to the first icon (that @nicolas-raoul liked as well).

Could anyone volunteer to send WMF Legal an email about this?

**Comment by debpedrano** [comment]
@misaochan  so you mean we will just stick on the base logo and put some modification to add some enhancement? I will do that if that's what you meant.

**Comment by sivaraam** [comment]
> Could anyone volunteer to send WMF Legal an email about this?

Hi @misaochan, just for the clarification: can anyone not much involved in the project also ask WMF legal about this? If that's the case I might help you with this.

**Comment by sivaraam** [comment]
> @sivaraam Sure, absolutely, please just link them to #868 (comment) , thanks :)

No problem :-) Just for the confirmation is [trademarks@wikimedia.org](mailto:trademarks@wikimedia.org) the email address to which I would have to send an email or is it [legal@wikimedia.org](mailto:legal@wikimedia.org) ?

**Comment by misaochan** [comment]
I think legal@wikimedia.org should work, if they need to forward it to trademarks they could do so?

**Comment by misaochan** [comment]
Thanks @sivaraam !

**Comment by debpedrano** [comment]
@misaochan  yes, I don't have work for now and its kinda bored to stay at home and do nothing that's why I am making issue (logo designs) to entertain myself. If you have any project that needs a logo, you can count on me, everything is for free.

**Comment by misaochan** [comment]
@debpedrano Ah, cool. Are you open to doing any other types of design, or do you only do logos?

**Comment by debpedrano** [comment]
@misaochan I am still a beginner, I only do logo(s). hoping soon I can do animation and ui design.

**Comment by nicolas-raoul** [comment]
@debpedrano It is not logos, but you might be interested in this project: https://github.com/nicolas-raoul/Icons-for-Wikidata :-)

**Comment by sivaraam** [comment]
> Let's wait for the response.

I glad to inform people here that WMF legal is fine with [icon that has been suggested](https://github.com/commons-app/apps-android-commons/issues/868#issuecomment-414520069) (the first one)! They're also fine with making that icon adaptive as long as only the background shape changes.

**Comment by misaochan** [comment]
Yay! @sdabhi23 Please feel free to submit your PR with the first icon. :)

**Comment by sdabhi23** [comment]
@misaochan as there are new year celebrations going on here (in India), will it be ok if I do it after a couple of days?

**Comment by sdabhi23** [comment]
Yeah. Thanks :D

**Comment by neslihanturan** [comment]
If we have some volunteer designers, I think nearby descriptor images can be a good point to start. 
![image](https://user-images.githubusercontent.com/3127881/48265313-42d20c00-e43d-11e8-922c-e15bea793852.png)

I like question mark but, when it comes to other icons, I think they don't match with our app style. Something similar to designs in achivements activity would be nice:
![image](https://user-images.githubusercontent.com/3127881/48265407-80cf3000-e43d-11e8-85a9-377ed110b7ee.png)

And using our base colors and having a consistent feeling between these icons could be good. 

Note: If you start to work on this task, please share step by step improvements.

**Comment by sdabhi23** [comment]
Hey is there a svg available for the logo for the beta  version? Or even a png would work!

**Comment by sdabhi23** [comment]
I would like to suggest using [this logo](https://commons.wikimedia.org/wiki/File:Commons-Beta-logo.svg) for the beta version.

![beta](https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Commons-Beta-logo.svg/178px-Commons-Beta-logo.svg.png)

**Comment by misaochan** [comment]
Ah, we actually have a logo for the beta version! Could you please take a look in the /res/ folder for it? I personally like your new logo (very ingenious, by the way ;)), but that would require us to ask permission from Legal all over again. Whereas if you use the existing beta logo and just modify the background similar to what you did for the Commons logo, we should be fine.

**Comment by nicolas-raoul** [comment]
The new icon is very small compared to before:
![screenshot_20181217-200237](https://user-images.githubusercontent.com/99590/50083525-6dcb2f00-0237-11e9-9f1d-3eaba9489319.png)
Would someone mind checking why?
It should be as big as the Alarm Klock (also open source so feel free to have a look at their configuration) icon.

**Comment by sdabhi23** [comment]
@nicolas-raoul can you share the configuration of you  device (like screen size, android version, screen resolution, etc)? I am not able to reproduce it.

Also keep in mind the fact that alarm klock is a legacy project with ant builds which is no longer being maintained, so it might behave oddly on some devices. The icons for our app were generated automatically by Android Studio from the background and foreground layers.

You can refer the image attached below for comparison with system app icons on Android 8.1, with adaptive icons:

![comparison image](https://user-images.githubusercontent.com/25759344/48956605-f4af2380-ef79-11e8-976d-c7c1854829c3.png)

**Comment by nicolas-raoul** [comment]
LineageOS 7.1.2 on Nexus 5 (1080×1920 pixels)
More icons for comaprison:
![screenshot_20181217-233508](https://user-images.githubusercontent.com/99590/50093652-a7f6f980-0254-11e9-867d-222629c1b6f0.png)

**Comment by domdomegg** [comment]
Got same issue as @nicolas-raoul 

LineageOS 7.1.2 on Samsung Galaxy S6 (2560x1440 pixels)

Evie launcher

**Screenshot**
![screenshot_20181219-003226](https://user-images.githubusercontent.com/4953590/50191508-1b1c7080-0326-11e9-816b-bae5b8bb6b86.png)

Also personally I prefer circular icons, which the app previously used. Is there any way to set that as the default icon style for devices that don't support adaptive icons (i.e. those with API level <= 25)? However probably up for debate as I think Wikipedia app does use rounded square.

**Comment by nicolas-raoul** [comment]
The icon size difference is especially visible in the Share dialog (please disregard the peppers), both icons are rounded squares but the "LINE" icon is much bigger:

![screenshot_20181220-142711](https://user-images.githubusercontent.com/99590/51311209-ba4e9800-1a8b-11e9-8a8c-e0e26f67c5e2.png)

**Comment by sivaraam** [comment]
I'm closing this as I think the main outcome of this issue has been achieved (choosing an icon that's visible in dark backgrounds).

@nicolas-raoul I believe you would be fine with opening a separate issue for the icon size difference if you feel it's still the case 🙂
